exonum-java-binding-core/rust/src/utils/errors.rs

```diff
@@ -46,6 +48,35 @@ pub fn check_error_on_exception<T>(env: &JNIEnv, result: JniResult<T>) -> Result
     })
 }
 
+/// Handles exceptions after executing transactions
+///
+/// The TransactionExecutionException and its descendants are converted into `Error`s with their
+/// descriptions. The rest (Java and JNI errors) are treated as unrecoverable and result in a panic.
+///
+/// Panics:
+/// - Panics if there is some JNI error.
+/// - If there is a pending Java throwable that IS NOT an instance or subclass of the
+/// `TransactionExecutionException`.
+pub fn check_transaction_execution_result<T>(
+    env: &JNIEnv,
+    result: JniResult<T>,
+) -> Result<T, String> {
+    result.map_err(|jni_error| match jni_error.0 {
+        JniErrorKind::JavaException => {
+            let exception = get_and_clear_java_exception(env);
+            let message = describe_java_exception(env, exception);
```

Comment:
I think we shall get the exception message instead of using `describe_java_exception` which includes the class name and is vaguely defined.